pydm/widgets/timeplot.py

```diff
@@ -231,24 +364,68 @@ def setCurves(self, new_list):
 
     curves = Property("QStringList", getCurves, setCurves)
 
+    def findCurve(self, pv_name):
+        for curve in self._curves:
+            if curve.address == pv_name:
+                return curve
+
+    def refreshCurve(self, curve):
+        """
+        Remove a curve currently being plotted on the timeplot, then redraw that curve, which could have been updated
+        with a new symbol, line style, line width, etc.
+        :param curve:
+        :return:
+        """
+        curve = self.findCurve(curve.channel)
+        if curve:
+            self.removeYChannel(curve)
+            self.addYChannel(y_channel=curve.address, color=curve.color, name=curve.address,
+                             lineStyle=curve.lineStyle, lineWidth=curve.lineWidth, symbol=curve.symbol,
+                             symbolSize=curve.symbolSize)
+
+    def annotateCurve(self, curve, annotation):
+        curve_point = CurvePoint(curve)
+        self.plotItem.addItem(curve_point)
+        annotation.setParentItem(curve_point)
+        curve_point.setPos(1)
+
+    def addLegendItem(self, item, pv_name, force_show_legend=False):
+        self._legend.addItem(item, pv_name)
+        self.setShowLegend(force_show_legend)
+
+    def removeLegendItem(self, pv_name):
+        self._legend.removeItem(pv_name)
+        if len(self._legend.items) == 0:
+            self.setShowLegend(False)
+
     def getBufferSize(self):
         return int(self._bufferSize)
 
     def setBufferSize(self, value):
         if self._bufferSize != int(value):
-            self._bufferSize = max(int(value), 1)
+            # Originally, the bufferSize is the max between the user's input and 1, and 1 doesn't make sense.
+            # So, I'm comparing the user's input with the minimum buffer size, and pick the max between the two
+            self._bufferSize = max(int(value), MINIMUM_BUFFER_SIZE)
```

Comment:
Buffer size of 1 can make sense if you are monitoring a single value across time... 
I believe that `MINIMUM_BUFFER_SIZE` can be one...
What do you think @mattgibbs

Reply:
No, a buffer size of 1 doesn't make much sense.  The buffer size is the size of the ring buffer that each curve uses to store its data.  A size of 1 would mean that the plot only ever shows the latest data point, all other data is thrown away.  I'd say the *minimum* buffer size is technically 2 - you need two points to draw a line.

Reply:
@mattgibbs what about the cases of BPM screen for instance (which would use the ScatterPlot) in which you look for the current position of a scalar?
I see the reason why not do it but it looks like an arbitrary decision... If someone wants to monitor a single point scalar across time in a graph and it makes sense for their use case, why not?

Reply:
I agree for ScatterPlot, but this is TimePlot, not ScatterPlot.  In TimePlot, if you set the buffer size to 1, you will just see a single dot that moves up and down on the right edge of the plot - no data except the current value will ever be shown.

Reply:
Agreed